src/app/zap-templates/endpoint_config.zapt

```diff
@@ -3,4 +3,117 @@
+#if BIGENDIAN_CPU
+#define GENERATED_DEFAULTS {{endpoint_attribute_long_defaults endian="big"}}
+
+#else // !BIGENDIAN_CPU
+#define GENERATED_DEFAULTS {{endpoint_attribute_long_defaults endian="little"}}
+#endif // BIGENDIAN_CPU
+
+#define GENERATED_DEFAULTS_COUNT ({{endpoint_attribute_long_defaults_count}})
+
+#define ZAP_TYPE(type) ZCL_ ## type ## _ATTRIBUTE_TYPE
+#define ZAP_LONG_DEFAULTS_INDEX(index) {(uint8_t*)(&generatedDefaults[index])}
+#define ZAP_MIN_MAX_DEFAULTS_INDEX(index) {(uint8_t*)(&minMaxDefault[index])}
+
+// This is an array of EmberAfAttributeMinMaxValue structures.
+#define GENERATED_MIN_MAX_DEFAULT_COUNT {{endpoint_attribute_min_max_count}}
+#define GENERATED_MIN_MAX_DEFAULTS {{endpoint_attribute_min_max_list}}
+
+#define ZAP_ATTRIBUTE_MASK(mask) ATTRIBUTE_MASK_ ## mask
+// This is an array of EmberAfAttributeMetadata structures.
+#define GENERATED_ATTRIBUTE_COUNT {{endpoint_attribute_count}}
+#define GENERATED_ATTRIBUTES {{ endpoint_attribute_list }}
+
+// This is an array of EmberAfCluster structures.
+#define ZAP_ATTRIBUTE_INDEX(index) ((EmberAfAttributeMetadata *)(&generatedAttributes[index]))
+
+#define ZAP_CLUSTER_MASK(mask) CLUSTER_MASK_ ## mask
+#define GENERATED_CLUSTER_COUNT {{endpoint_cluster_count}}
+#define GENERATED_CLUSTERS {{endpoint_cluster_list}}
+
+#define ZAP_CLUSTER_INDEX(index) ((EmberAfCluster*)(&generatedClusters[index]))
+
+
+// This is an array of EmberAfEndpointType structures.
+#define GENERATED_ENDPOINT_TYPES {{endpoint_types_list}}
+
+
+// Largest attribute size is needed for various buffers
+#define ATTRIBUTE_LARGEST ({{endpoint_largest_attribute_size}})
+
+// Total size of singleton attributes
+#define ATTRIBUTE_SINGLETONS_SIZE ({{endpoint_singletons_size}})
+
+// Total size of attribute storage
+#define ATTRIBUTE_MAX_SIZE ({{endpoint_total_storage_size}})
+
+// Number of fixed endpoints
+#define FIXED_ENDPOINT_COUNT ({{endpoint_count}})
+
+// Array of endpoints that are supported, the data inside
+// the array is the endpoint number.
+#define FIXED_ENDPOINT_ARRAY {{endpoint_fixed_endpoint_array}}
+
+// Array of profile ids
+#define FIXED_PROFILE_IDS {{endpoint_fixed_profile_id_array}}
+
+// Array of device ids
+#define FIXED_DEVICE_IDS {{endpoint_fixed_device_id_array}}
+
+// Array of device versions
+#define FIXED_DEVICE_VERSIONS {{endpoint_fixed_device_version_array}}
+
+// Array of endpoint types supported on each endpoint
+#define FIXED_ENDPOINT_TYPES {{endpoint_fixed_endpoint_type_array}}
+
+// Array of networks supported on each endpoint
+#define FIXED_NETWORKS {{endpoint_fixed_network_array}}
+
+
+// Array of EmberAfCommandMetadata structs.
+#define ZAP_COMMAND_MASK(mask) COMMAND_MASK_ ## mask
+#define EMBER_AF_GENERATED_COMMAND_COUNT ({{endpoint_command_count}})
+#define GENERATED_COMMANDS {{endpoint_command_list}}
+
+// Array of EmberAfManufacturerCodeEntry structures for commands.
+#define GENERATED_COMMAND_MANUFACTURER_CODE_COUNT ({{endpoint_command_manufacturer_code_count}})
```

Comment:
I'm pretty sure it does not work. I have recently added a `MfgSpecificPing` method to chip, and generating this file with it does not increment this number.

Reply:
Fixed